clients/git/client.go

```diff
@@ -188,17 +201,58 @@ func (c *Client) Search(request clients.SearchRequest) (clients.SearchResponse,
 	return ret, nil
 }
 
-// TODO(#1709): Implement below fns using go-git.
-func (c *Client) SearchCommits(request clients.SearchCommitsOptions) ([]clients.Commit, error) {
-	return nil, nil
-}
-
 func (c *Client) ListFiles(predicate func(string) (bool, error)) ([]string, error) {
-	return nil, nil
+	var files []string
```

Comment:
I think as written this is fine. Feel free to resolve, just leaving breadcrumbs. In the future, I'd be curious if an alternative pattern is useful/better:
I was looking at this for a related issue (#1499), and go-git uses a slightly different pattern in their examples:

https://github.com/go-git/go-git/blob/77264da94ee6090b759ed5763d84302619cc482f/_examples/showcase/main.go#L37-L57

Reply:
I never thought of this as an option. This is really cool. I am fixing it. Thank you!